datafusion/core/tests/sql/idenfifers.rs

```diff
@@ -0,0 +1,226 @@
+    let expected = vec![
+        "+---+---+",
+        "| x | b |",
+        "+---+---+",
+        "| 1 | 2 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+
+    let sql = r#"SELECT t.A AS "X", b FROM case_insensitive_test AS t"#;
+    let result = plan_and_collect(&ctx, sql)
+        .await
+        .expect("ran plan correctly");
+    let expected = vec![
+        "+---+---+",
+        "| X | b |",
+        "+---+---+",
+        "| 1 | 2 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+
+    // Order by
+
+    let sql = "SELECT t.A AS x, b FROM case_insensitive_test AS t ORDER BY x";
+    let result = plan_and_collect(&ctx, sql)
+        .await
+        .expect("ran plan correctly");
+    let expected = vec![
+        "+---+---+",
+        "| x | b |",
+        "+---+---+",
+        "| 1 | 2 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+
+    let sql = "SELECT t.A AS x, b FROM case_insensitive_test AS t ORDER BY X";
+    let result = plan_and_collect(&ctx, sql)
+        .await
+        .expect("ran plan correctly");
+    let expected = vec![
+        "+---+---+",
+        "| x | b |",
+        "+---+---+",
+        "| 1 | 2 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+
+    let sql = r#"SELECT t.A AS "X", b FROM case_insensitive_test AS t ORDER BY "X""#;
+    let result = plan_and_collect(&ctx, sql)
+        .await
+        .expect("ran plan correctly");
+    let expected = vec![
+        "+---+---+",
+        "| X | b |",
+        "+---+---+",
+        "| 1 | 2 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+
+    // Where
+
+    let sql = "SELECT a, b FROM case_insensitive_test where A IS NOT null";
+    let result = plan_and_collect(&ctx, sql)
+        .await
+        .expect("ran plan correctly");
+    let expected = vec![
+        "+---+---+",
+        "| a | b |",
+        "+---+---+",
+        "| 1 | 2 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+
+    // Group by
+
+    let sql = "SELECT a as x, count(*) as c FROM case_insensitive_test GROUP BY X";
+    let result = plan_and_collect(&ctx, sql)
+        .await
+        .expect("ran plan correctly");
+    let expected = vec![
+        "+---+---+",
+        "| x | c |",
+        "+---+---+",
+        "| 1 | 1 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+
+    let sql = r#"SELECT a as "X", count(*) as c FROM case_insensitive_test GROUP BY "X""#;
+    let result = plan_and_collect(&ctx, sql)
+        .await
+        .expect("ran plan correctly");
+    let expected = vec![
+        "+---+---+",
+        "| X | c |",
+        "+---+---+",
+        "| 1 | 1 |",
+        "+---+---+",
+    ];
+    assert_batches_sorted_eq!(expected, &result);
+}
+
+#[tokio::test]
+async fn case_insensitive_in_sql_errors() {
```

Comment:
This is a new test added in this PR, based on the one from @ReggieFan on https://github.com/apache/arrow-datafusion/pull/2211 to make it clear this is expected behavior